enterprise/backend/src/metabase_enterprise/cache/strategies.clj

```diff
@@ -28,64 +26,52 @@
+                [:invalidated-at {:optional true} some?]]]]])
 
 ;;; Querying DB
 
-(defn granular-duration-hours
-  "Returns the granular cache duration (in hours) for a card. On EE, this first checking whether there is a stored value
-   for the card, dashboard, or database (in that order of decreasing preference). Returns nil on OSS."
-  [card dashboard-id]
-  (or (:cache_ttl card)
-      (when dashboard-id
-        (t2/select-one-fn :cache_ttl [:model/Dashboard :cache_ttl] :id dashboard-id))
-      (t2/select-one-fn :cache_ttl [:model/Database :cache_ttl] :id (:database_id card))))
-
-(defenterprise-schema granular-cache-strategy :- [:maybe (CacheStrategy)]
+(defenterprise-schema cache-strategy :- [:maybe (CacheStrategy)]
   "Returns the granular cache strategy for a card."
   :feature :cache-granular-controls
   [card dashboard-id]
-  (when (public-settings/enable-query-caching)
-    (let [qs   (for [[i model model-id] [[1 "question"   (:id card)]
-                                         [2 "dashboard"  dashboard-id]
-                                         [3 "database"   (:database_id card)]
-                                         [4 "root"       0]]
-                     :when              model-id]
-                 {:from   [:cache_config]
-                  :select [:id
-                           [[:inline i] :ordering]]
-                  :where  [:and
-                           [:= :model model]
-                           [:= :model_id model-id]]})
-          q    {:from     [[{:union-all qs} :unused_alias]]
-                :select   [:id]
-                :order-by :ordering
-                :limit    1}
-          item (t2/select-one :model/CacheConfig :id q)]
-      (if item
-        (-> (:strategy (api.cache/row->config item))
-            (m/assoc-some :invalidated-at (t/max (:invalidated_at item)
-                                                 (:cache_invalidated_at card))))
-        (when-let [ttl (granular-duration-hours card dashboard-id)]
-          {:type     :duration
-           :duration ttl
-           :unit     "hours"})))))
+  (let [qs   (for [[i model model-id] [[1 "question"   (:id card)]
+                                       [2 "dashboard"  dashboard-id]
+                                       [3 "database"   (:database_id card)]
+                                       [4 "root"       0]]
+                   :when              model-id]
+               {:from   [:cache_config]
+                :select [:id
+                         [[:inline i] :ordering]]
+                :where  [:and
+                         [:= :model [:inline model]]
+                         [:= :model_id model-id]]})
+        q    {:from     [[{:union-all qs} :unused_alias]]
+              :select   [:id]
+              :order-by :ordering
+              :limit    [:inline 1]}
+        item (t2/select-one :model/CacheConfig :id q)]
+    (when item
+      (:strategy (cache-config/row->config item card)))))
```

Comment:
Rather than overloading `row->config`, perhaps you could replace "directly reaching inside" with a method that captures how you are applying a potentially more general config to the strategy for a specific card. 

```suggestion
      (card-strategy (cache-config/row->config item) card))))
```

This also leaves `row->config` as one part of the database bijection, keeping the semantics nice and clean.

Reply:
You've persuaded me, done :)